### PR TITLE
DAOS-15841 control: Improve coverage and fs mocking in storage packages

### PR DESCRIPTION
**Pull request by tanabarr**
Some control-plane storage unit tests are inadvertently calling into
test runner host OS filesystem. Fix by consolidating SystemProvider
interface and applying storage subsystem provider stubs by default
in the unit test framework. As a result coverage can be improved by
exercising a greater number of code paths.

Features: control
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by mjmac — DISMISSED** [review]

File: src/control/provider/system/mocks.go [hunk 1]
```diff
@@ -44,10 +44,19 @@ type (
...
 		GetfsUsageResps []GetfsUsageRetval
-		GetFsTypeRes    *FsType
-		GetFsTypeErr    []error
+		GetfsTypeRes    *FsType
+		GetfsTypeErr    []error
 		StatErrors      map[string]error
 		RealStat        bool
+		ReadFileResults map[string][]byte
+		ReadFileErrors  map[string]error
+		RealReadFile    bool
```
Comment (kjacque) [inline src/control/provider/system/mocks.go:53]: I see `RealStat` already exists up there, but I don't feel great about adding this kind of complexity to a single mock struct, to toggle the real version of the function on and off. IMO, if there are certain tests that want some real and some mocked functions, those tests should create their own mock struct to customize the behavior.

**Review by kjacque — APPROVED** [review]
Wouldn't block on my comment, but it is something to think about. Everything else looks okay.

**Review by knard-intel — APPROVED** [review]
LGTM

File: src/control/provider/system/mocks.go (hunk 1 above)
Comment (knard-intel) [reply src/control/provider/system/mocks.go:53]: From my side, it has the advantage to factorize this mocking code and the complexity seems to still be acceptable.
However, as I am not a GO expert, I am probably missing something.

Comment (tanabarr) [reply src/control/provider/system/mocks.go:53]: @knard-intel those were my thoughts when adding the flag. After chatting with @kjacque I can see benefit in moving to smaller package-local (as @mjmac also suggested elsewhere) SystemProvider mocks which would be consistent with having package-level SystemProvider interfaces. I will make changes if I have to repush or the next time I visit this area.

Comment (knard-intel) [reply src/control/provider/system/mocks.go:53]: Make sense, thanks for the explanation :-)

**Review by mjmac — APPROVED** [review]

**Pull request merged**